### ISSUE TITLE
getSubscriptions empty after publishing (App Rejected)

### ISSUE THREAD

**Issue by fcandiani** [dooboolab/react-native-iap#644]
### Version of react-native-iap
"react-native-iap": "^3.3.9"

### Version of react-native
"react-native": "0.59.8",

### Platforms you faced the error (IOS or Android or both?)
IOS 

### Behavior
When testing in local environment subscriptions are shown and available for purchase, but when published to the app store they say that they are no products listed (screenshot attached)

### Tested environment (Emulator? Real Device?)
Emulator, Real Device

Code (few parts omitted for simplicity)

```
import RNIap, {
	purchaseUpdatedListener,
	purchaseErrorListener,
} from 'react-native-iap';

const itemSubs = Platform.select({
	ios: [
		'br.com.pontue.correcao.uma',
		'br.com.pontue.correcao.duas',
		'br.com.pontue.correcao.quatro',
               'br.com.pontue.correcao.men'
	]
});

async componentDidMount() {
	try {
		const result = await RNIap.initConnection();
		const products = await RNIap.getSubscriptions(itemSubs);
		this.setState({ loading: false, productList: products });
		console.log(this.state.productList)
	} catch (err) {
		console.warn(err.code, err.message);
	}
	
	purchaseUpdateSubscription = purchaseUpdatedListener(async(purchase) => {
		console.log('purchaseUpdatedListener', purchase);
		const receipt = purchase.transactionReceipt;
		if (receipt) {
			this.setState({ purchase }, () => this.showReceipt());
		}
	});

	purchaseErrorSubscription = purchaseErrorListener((error) => {
		console.log('purchaseErrorListener', error);
	});
}

componentWillMount() {
	if (purchaseUpdateSubscription) {
		purchaseUpdateSubscription.remove();
		purchaseUpdateSubscription = null;
	}
	if (purchaseErrorSubscription) {
		purchaseErrorSubscription.remove();
		purchaseErrorSubscription = null;
	}
}

requestSubscription = async sku => {
	try {
		await RNIap.requestSubscription(sku);
	} catch (err) {
		console.warn(err.code, err.message);
	}
}

render() {
       
![Captura de Tela 2019-08-14 às 18 22 11](https://user-images.githubusercontent.com/7735185/63057399-7f490180-bec0-11e9-9046-9f461317947f.png)
return this.state.productList.map((product, i) => {
		return (
			<PurchasableItem key={i}
				title={itemTitle[product.productId]}
				currency='R$'
				price={price[product.productId]}
				cents={`,${cents[product.productId]}`}
				subtitle={itemDesc[product.productId]}
				doPurchase={this.requestSubscription.bind(this)}
				id={product.productId}
			/>
		)
	})
}
```

Screenshots
![Screenshot from 2019-08-14 17-51-51](https://user-images.githubusercontent.com/7735185/63056822-3e042200-bebf-11e9-83ac-36b477e457ad.png)
![Screenshot from 2019-08-14 18-12-03](https://user-images.githubusercontent.com/7735185/63056863-54aa7900-bebf-11e9-99af-1fb10df2b228.png)
![Captura de Tela 2019-08-14 às 18 22 11](https://user-images.githubusercontent.com/7735185/63057451-a3a4de00-bec0-11e9-91a4-7375c6f17745.png)

Apple Screenshot (added in the resolution center)
https://iosapps-ssl.itunes.apple.com/itunes-assets/Purple123/v4/83/3b/69/833b6983-35ad-388f-3877-a59a7c069ec3/attachment-3102671694687178061Screenshot-0813-135321.png?accessKey=1566076151_2113190433823191526_KUpsUTx8uu27FJFVw%2FktAW%2F8Ce%2FTuMT5%2FTMfgJhSw2oK%2BdvePAiC7DqcghmecZSMDblIsoEvruxvwVu0cL4k7XJ6E3gbT1EGlgiA7zLTvCsMXYrQzIK8iewq7qvMe%2FTcGrbjm4KNzjz%2FL66zEeKWz73ofyLXzczB32LUmcJh%2BrmuUOMy02%2Fkz6tXLc3cmFmzxTgKx%2Fs5sF8JXq3TRZhavQ%3D%3D

**Comment by jvandenaardweg** [comment]
Can you show your whole component? I'm curious what the default state of `productList` is. If it's not `[]` there already might be a rendering error when you map over `this.state.productList` upon first render: `Cannot read property 'map' of`

Also, does it work for you when you test your release build using TestFlight?

**Comment by fcandiani** [comment]
I'm not using TestFlight (I don't have an iphone/ipad of my own, when I tested on a real device I was using a friends phone).

The default state for productList is []

I tested using release build as a config in xcode

`Go to Product -> Scheme -> Edit Scheme in XCode. Select Run tab and set the Build Configuration dropdown to Release.`

Attached is all the code used in the iap flow as well as package.json
[iap.zip](https://github.com/dooboolab/react-native-iap/files/3505471/iap.zip)

**Comment by jvandenaardweg** [comment]
Have you tested it on an iPad simulator? As the reviewer seems to have used an iPad. Maybe it's a layout issue. I don't see anything weird in your code

**Comment by fcandiani** [comment]
It works normally, release and debug mode

![Captura de Tela 2019-08-15 às 10 22 23](https://user-images.githubusercontent.com/7735185/63097374-9fbb9f00-bf46-11e9-9f35-ba84e10f332e.png)

**Comment by hyochan** [comment]
@fcandiani It looks like you've missed something when publishing your app. Have you correctly checked the `IAP` products that should be published with your app? Previously, someone missed out some steps and had the same issue. You should also check this with `apple` support.

**Comment by fcandiani** [comment]
Just something I just notice is that the product returned in dev is quite different from what is defined in 

index.d.ts <18-34>
```
export interface Subscription<ID extends string> extends Common {
  type: 'subs' | 'sub';
  productId: ID;

  introductoryPrice?: string;
  introductoryPricePaymentModeIOS?: string;
  introductoryPriceNumberOfPeriodsIOS?: string;
  introductoryPriceSubscriptionPeriodIOS?: string;

  subscriptionPeriodNumberIOS?: string;
  subscriptionPeriodUnitIOS?: string;

  introductoryPriceCyclesAndroid?: string;
  introductoryPricePeriodAndroid?: string;
  subscriptionPeriodAndroid?: string;
  freeTrialPeriodAndroid?: string;
}
```

Returned value

```
(3) [{…}, {…}, {…}]
    0:
    currency: "BRL"
    description: ""
    discounts: []
    introductoryPrice: ""
    introductoryPriceNumberOfPeriodsIOS: ""
    introductoryPricePaymentModeIOS: ""
    introductoryPriceSubscriptionPeriodIOS: ""
    localizedPrice: "R$19,90"
    price: "19.9"
    productId: "br.com.pontue.correcao.duas"
    subscriptionPeriodNumberIOS: "1"
    subscriptionPeriodUnitIOS: "MONTH"
    title: ""
    type: "Do not use this. It returned sub only before"
    __proto__: Object
```

**Comment by fcandiani** [comment]
@hyochan You mean adding the product when publishing a new release? Yes, I did at the first attempt (as stated in https://forums.developer.apple.com/thread/26864 and https://forums.developer.apple.com/thread/92214 there is no need to redo it after).

I'm going to try to make a new release today and ask for a call to follow the process (a friend of mine told me this is possible - still have to check how)

**Comment by fcandiani** [comment]
@hyochan I managed to figure out what was wrong. The app was submitted and then rejected a few times. After I made some changes to my code to fix it (not related to iap) I changed the products names to resubmit it. What I forgot to have a look was the subscription group, which was "rejected" as well (having a red small circle before its language configuration). Yesterday I resubmitted with that in yellow circle (made a name change, saved and restored the change again) and they were able to run it.


I was thinking, since I had quite a headache with publishing it to apple store, to make an pr to be added to the documentation (or a link to it at least) of the process that I had to follow with apple store connect and my code. So, in the future.

If that is ok, I would like a little bit of guidence on how to do that correctly.

Thanks for the help
